File: SWIG/Examples/test-suite/python/director_string_runme.py
from director_string import *

class B(A):
  def __init__(self,string):
    A.__init__(self,string)

  def get_first(self):
    return A.get_first(self) + " world!"
    


b = B("hello")

b.get(0)
if b.get_first() != "hello world!":
  print b.get_first() 
  raise RuntimeError

File: Examples/test-suite/perl5/director_string_runme.pl
use strict;
use warnings;
use director_string;

{
	package B;
	use base 'director_string::A';
	our $in_first = 0;
	sub get_first { my($self) = @_;
		die "SUPER RESOLVE BAD" if $in_first;
		local $in_first = 1;
		return $self->SUPER::get_first() . " world!";
	}
	our %smem;
	our $in_process_text = 0;
	sub process_text { my($self, $string) = @_;
		die "SUPER RESOLVE BAD" if $in_process_text;
		local $in_process_text = 1;
		$self->SUPER::process_text($string);
		$smem{$self} = "hello";
	}
}

my $b = B->new("hello");

$b->get(0);

die "RuntimeError" if
	$b->get_first() ne "hello world!";

$b->call_process_func();

die "RuntimeError" if $B::smem{$b} ne "hello";
